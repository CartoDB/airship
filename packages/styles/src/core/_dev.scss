@import 'tools/_mixins';
@import 'variables/_variables';
