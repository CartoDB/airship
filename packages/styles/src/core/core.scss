@import 'variables/_variables';

* {
  box-sizing: border-box;
  font-family: sans-serif;
}

:root {
  --primary-color: #{$primary-color};
}
