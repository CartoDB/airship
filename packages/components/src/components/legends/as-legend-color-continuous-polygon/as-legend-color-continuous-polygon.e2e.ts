import { E2EPage, newE2EPage } from '@stencil/core/testing';

describe('as-legend-color-continuous-polygon', async () => {
  describe('Rendering', async () => {
    let page: E2EPage;

    it('basic', async () => {
      page = await newE2EPage({ html: `
        <as-legend-color-continuous-polygon>
        </as-legend-color-continuous-polygon>
      ` });

      const element = await page.find('as-legend-color-continuous-polygon');
      element.setProperty('data', [{
        color: '#FF0000',
        label: 'point label'
      },
      {
        color: '#00FF00',
        label: 'line label'
      },
      {
        color: '#0000FF',
        label: 'polygon label'
      }]);
      await page.waitForChanges();
      expect(element.outerHTML).toMatchSnapshot();
    });

    it('basic horizontal', async () => {
      page = await newE2EPage({ html: `
        <as-legend-color-continuous-polygon>
        </as-legend-color-continuous-polygon>
      ` });

      const element = await page.find('as-legend-color-continuous-polygon');
      element.setProperty('data', [{
        color: '#FF0000',
        label: 'point label'
      },
      {
        color: '#00FF00',
        label: 'line label'
      },
      {
        color: '#0000FF',
        label: 'polygon label'
      }]);
      element.setProperty('orientation', 'horizontal');
      await page.waitForChanges();
      expect(element.outerHTML).toMatchSnapshot();
    });

    it('should contain one item less than the number of data entries', async () => {
      page = await newE2EPage({ html: `
        <as-legend-color-continuous-polygon>
        </as-legend-color-continuous-polygon>
      ` });

      const element = await page.find('as-legend-color-continuous-polygon');
      element.setProperty('data', [{
        color: '#FF0000',
        label: 'point label'
      },
      {
        color: '#00FF00',
        label: 'line label'
      },
      {
        color: '#0000FF',
        label: 'polygon label'
      }]);
      await page.waitForChanges();

      const gradients = await page.findAll('.as-legend-color-continuous-polygon--step');
      expect(gradients.length).toEqual(2);
    });
  });
});
