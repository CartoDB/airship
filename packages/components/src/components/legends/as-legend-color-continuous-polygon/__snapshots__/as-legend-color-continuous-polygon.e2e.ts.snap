// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`as-legend-color-continuous-polygon Rendering basic 1`] = `"<as-legend-color-continuous-polygon class=\\"hydrated\\"> <div class=\\"as-legend-color-continuous-polygon--outer-wrapper as-legend-color-continuous-polygon--vertical\\"><div class=\\"as-legend-color-continuous-polygon--wrapper as-legend-color-continuous-polygon--color\\"><div class=\\"as-legend-color-continuous-polygon--step\\" style=\\"background-image: linear-gradient(rgb(255, 0, 0), rgb(0, 255, 0));\\"></div><div class=\\"as-legend-color-continuous-polygon--step\\" style=\\"background-image: linear-gradient(rgb(0, 255, 0), rgb(0, 0, 255));\\"></div></div><div class=\\"as-legend-color-continuous-polygon--wrapper as-legend-color-continuous-polygon--labels\\"><div class=\\"as-legend-color-continuous-polygon--label\\"><span>line label</span></div><div class=\\"as-legend-color-continuous-polygon--label\\"><span>polygon label</span></div></div></div></as-legend-color-continuous-polygon>"`;

exports[`as-legend-color-continuous-polygon Rendering basic horizontal 1`] = `"<as-legend-color-continuous-polygon class=\\"hydrated\\"> <div class=\\"as-legend-color-continuous-polygon--outer-wrapper as-legend-color-continuous-polygon--horizontal\\"><div class=\\"as-legend-color-continuous-polygon--wrapper as-legend-color-continuous-polygon--color\\"><div class=\\"as-legend-color-continuous-polygon--step\\" style=\\"background-image: linear-gradient(0.25turn, rgb(255, 0, 0), rgb(0, 255, 0));\\"></div><div class=\\"as-legend-color-continuous-polygon--step\\" style=\\"background-image: linear-gradient(0.25turn, rgb(0, 255, 0), rgb(0, 0, 255));\\"></div></div><div class=\\"as-legend-color-continuous-polygon--wrapper as-legend-color-continuous-polygon--labels\\"><div class=\\"as-legend-color-continuous-polygon--label\\"><span>point label</span></div><div class=\\"as-legend-color-continuous-polygon--label\\"><span>line label</span></div></div></div></as-legend-color-continuous-polygon>"`;
